### PR TITLE
Deep linking to item view when notification is clicked

### PR DESCRIPTION
**Pull request by arcturus**
@wilsonpage r?

Before we were just handling the click on the custom actions, but not the normal notification.

This will end fixing #189

### CODE REVIEW

File: ios/NotificationsHelperIOS10.swift [hunk 1]
```diff
@@ -85,6 +85,11 @@ class NotificationsHelperIOS10: NSObject, UNUserNotificationCenterDelegate {
       debugPrint("Launching web page \(response.notification.request.identifier)")
       let url = NSURL(string: response.notification.request.identifier)
       UIApplication.sharedApplication().openURL(url!, options: [:], completionHandler: nil)
+    } else {
+      // We were clicked but not the visit (go to browser) action
+      // so we navigate to the deep link inside the app
+      let url = NSURL(string: "mozilla-magnet:item?url=\(response.notification.request.identifier)")
```
Comment (wilsonpage) [inline ios/NotificationsHelperIOS10.swift]: Shouldn't this be: `mozilla-magnet://item?url=...`

**Comment by wilsonpage** [comment]
- The custom protocol looks wrong, it's missing the `//`
- Shouldn't we also be deep linking for < iOS-10?

**Comment by wilsonpage** [comment]
r+ after those comments addressed

File: ios/NotificationsHelperIOS10.swift (hunk 1 above)
Comment (arcturus) [reply ios/NotificationsHelperIOS10.swift]: Yeah you are right, it's working, but definitely better to add the //

**Comment by arcturus** [comment]
> Shouldn't we also be deep linking for < iOS-10?

Yes, thanks for the remind, it's pretty easy to do in the AppDelegate.
Good catch!

**Comment by arcturus** [comment]
Oh I just realised, for ios9 we say 'Content nearby' we don't show specific notifications for poi, so with that behaviour makes sense to send the user to the list. 

As well we don't know what the user wants to see, as that notification could mean that you found several things.

**Pull request merged**